src/client.js

```diff
@@ -1654,16 +1654,41 @@ class Client {
     async getSchema(schemaId, representation, internal) {
         var entity = await this._entityCache.get("xtk:schema", schemaId);
         if (!entity) {
-          entity = await this.getEntityIfMoreRecent("xtk:schema", schemaId, "xml", internal);
-          if (entity) {
-            const impls = DomUtil.getAttributeAsString(entity, "implements");
-            if (impls === "xtk:persist" && schemaId !== "xtk:session" && schemaId !== "xtk:persist") {
-                // Ensure xtk:persist is present by loading the xtk:session schema
-                await this.getSchema("xtk:session", "xml", true);
+              // special case of temp schema for nms:group
+              if (schemaId.startsWith("temp:group:")) {
+                  var parts = schemaId.split(":");
+                  var queryDef = {
+                    "schema": "nms:group",
+                    "operation": "get",
+                    "select": {
+                      "node": [
+                        { "expr": "@id" },
+                        { "expr": "extension" }
+                      ]
+                    },
+                    "where": {
+                      "condition": [
+                        { "expr": "@id=" + parts[2] }
+                      ]
+                    }
+                  };
+                  // Convert to current representation
+                  queryDef = this._convertToRepresentation(queryDef, "SimpleJson");
+                  const query = this.NLWS.xtkQueryDef.create(queryDef);
+                  const groupSchema = await query.executeQuery();
+                  entity = this._convertToRepresentation(groupSchema.extension, "SimpleJson", "xml");
+              } else {
+                  entity = await this.getEntityIfMoreRecent("xtk:schema", schemaId, "xml", internal);
+                  if (entity) {
+                      const impls = DomUtil.getAttributeAsString(entity, "implements");
+                      if (impls === "xtk:persist" && schemaId !== "xtk:session" && schemaId !== "xtk:persist") {
+                        // Ensure xtk:persist is present by loading the xtk:session schema
+                        await this.getSchema("xtk:session", "xml", true);
+                      }
+                      await this._entityCache.put("xtk:schema", schemaId, entity);
+                      await this._methodCache.put(entity);
+                  }
             }
-            await this._entityCache.put("xtk:schema", schemaId, entity);
```

Comment:
Is there any reason why the temp:group schemas are not cached?
If so, I think it's important to have a comment in the code that explains why

Reply:
because it fails when calling `put` method of `XtkEntityCache`
```js
const namespace = entity.getAttribute("namespace");
```

Reply:
I don't understand. I do not think that's a good reason. Having a different behaviour depending on the schema can be very comfusing and source of bugs, incorrect assumptions for the user. So if we do not cache such schemas, there should be a good reason. In other words we decide and justify : do we WANT to cache such schemas or not. If not, we explain and document. If yes, we make it work in all scenarios, and this means understnading which the put method fails in this case, and find a solution.

Reply:
I fix the put method failure, temp schemas are now cached